### PR TITLE
Move more logic to the crate

### PR DESCRIPTION
**Pull request by mkeeter**
In general, the goal of this PR is to refactor `lpc55_support`, making it more suitable for use as a dependency in `hubtools`.  This led to a bunch of somewhat-unrelated changes:

- Switching from using homebrew macros to `log` (with a tweaked logger) when checking certificate validity
- Move certificate validity checking from the `bin` to the crate; have it return an error as well as printing stuff
- Move `lpc55_sign` to a standalone crate `lpc55_sign_bin`, so that its dependencies don't tag along
- Switch from `anyhow` to `thiserror` for the library code (following [these best practices](https://nick.groenen.me/posts/rust-error-handling/#libraries-versus-applications))

### CODE REVIEW

File: lpc55_sign/src/signed_image.rs [hunk 1]
```diff
@@ -13,18 +13,40 @@ use rsa::{
...
     RsaPrivateKey, RsaPublicKey,
 };
+use serde::Deserialize;
 use sha2::{Digest, Sha256};
 use x509_parser::parse_x509_certificate;
 
-#[derive(Clone, Debug, Parser)]
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct CertConfig {
```
Comment (plotnick) [inline lpc55_sign/src/signed_image.rs:22]: I think this struct is only of use to the binary, so I might move it to `lpc55_sign_bin/src/main.rs`. By the time certs get into this library code, they should already be in memory.

File: lpc55_sign_bin/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                    .to_vec()?,
+                )?;
+                info!("CMPA written to {}", dest_cmpa.display());
+            }
+            if let Some(dest_cfpa) = &dest_cfpa {
+                std::fs::write(
+                    dest_cfpa,
+                    signed_image::generate_cfpa(
+                        debug_settings,
+                        [
```
Comment (plotnick) [inline lpc55_sign_bin/src/main.rs]: I'm fine hard-coding these bit values here, but might be worth a comment.

File: lpc55_sign_bin/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+            help = "pin on which to indicate boot errors"
+        )]
+        boot_err_pin: u8,
+    },
+    Cfpa {
+        dest_cfpa: PathBuf,
+    },
+    /// Generate a secure signed image
+    #[clap(name = "signed-image", group = clap::ArgGroup::new("mode").multiple(false))]
+    SignedImage {
```
Comment (plotnick) [inline lpc55_sign_bin/src/main.rs:64]: Every time I use this command, I vaguely wish it were called `sign-image`.

File: lpc55_sign_bin/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    #[clap(short = 'i', long = "in", help = "source file (binary)")]
+    src_bin: PathBuf,
+    #[clap(short = 'o', long = "out", help = "output file (binary)")]
+    dest_bin: PathBuf,
+    #[clap(long, default_value_t = 0)]
+    address: u32,
+}
+
+#[derive(Debug, Parser)]
+enum Command {
```
Comment (plotnick) [inline lpc55_sign_bin/src/main.rs:29]: This new command set makes much more sense. Thanks for re-organizing again.

**Review by plotnick — APPROVED** [review]
This seems great. Thanks for making the verification routines callable from library code, too.

I have verified that this at least compiles with permission slip if I detach CMPA & CFPA generation. I will re-attach (probably through hubtools) when the necessary parameters are all properly tracked.

File: lpc55_sign/src/signed_image.rs (hunk 1 above)
Comment (mkeeter) [reply lpc55_sign/src/signed_image.rs:22]: This was _originally_ in the binary, but I realized that it was duplicating some code in Hubris, so I consolidated it here:
https://github.com/oxidecomputer/hubris/blob/hubtools-signing/build/xtask/src/config.rs#L567

Comment (plotnick) [reply lpc55_sign/src/signed_image.rs:22]: I see. That makes sense, thanks.

File: lpc55_sign_bin/src/main.rs (hunk 3 above)
Comment (plotnick) [reply lpc55_sign_bin/src/main.rs]: Thanks!

**Comment by mkeeter** [comment]
Thanks for the review!  I'm going to do a final test and make sure that Hubris images are identical before merging, probably first thing tomorrow.

**Comment by mkeeter** [comment]
Hubris images aren't identical, alas, because this bumped some dependencies; still, `stage0` works on my bench, so I think we can merge this in.

**Pull request merged**